### PR TITLE
[AWS EKS - Scale-to-0] Add check for the AWS EKS tags on the ASG

### PR DESCRIPTION
**Pull request by MyannaHarris**
This change is the first change for the AWS EKS Managed Nodegroups support for scale-to-0 changes in the cluster autoscaler. It checks for the AWS EKS specific tags that we automaticaly add for Managed Nodegroups.

This change also adds unit tests for the new tag labels.

Example of the "Nodegroup %s is an EKS managed nodegroup." log line being shown with a log verbosity of --v=5

```
I0903 00:10:06.770366       1 filter_out_schedulable.go:157] Pod default.static-web marked as unschedulable can be scheduled on node template-node-for-eks-test-cas-ng-gpu-d8bdd47d-8232-234e-0544-7d7b3f16519d-8999011805617471788-upcoming-0. Ignoring in scale up.
I0903 00:10:06.770378       1 filter_out_schedulable.go:170] 0 pods were kept as unschedulable based on caching
I0903 00:10:06.770383       1 filter_out_schedulable.go:171] 1 pods marked as unschedulable can be scheduled.
I0903 00:10:06.770391       1 filter_out_schedulable.go:79] Schedulable pods present
I0903 00:10:06.770405       1 static_autoscaler.go:420] No unschedulable pods
I0903 00:10:06.770418       1 static_autoscaler.go:467] Calculating unneeded nodes
I0903 00:10:06.770434       1 scale_down.go:509] Scale-down calculation: ignoring 1 nodes unremovable in the last 5m0s
I0903 00:10:06.770457       1 static_autoscaler.go:521] Scale down status: unneededOnly=true lastScaleUpTime=2021-09-03 00:09:46.568646087 +0000 UTC m=+349.162905885 lastScaleDownDeleteTime=2021-09-02 23:04:24.733880672 +0000 UTC m=-3572.671859455 lastScaleDownFailTime=2021-09-02 23:04:24.733880672 +0000 UTC m=-3572.671859455 scaleDownForbidden=true isDeleteInProgress=false scaleDownInCooldown=true
I0903 00:10:07.037416       1 leaderelection.go:278] successfully renewed lease kube-system/cluster-autoscaler
I0903 00:10:09.060745       1 leaderelection.go:278] successfully renewed lease kube-system/cluster-autoscaler
I0903 00:10:11.082494       1 leaderelection.go:278] successfully renewed lease kube-system/cluster-autoscaler
I0903 00:10:13.094924       1 leaderelection.go:278] successfully renewed lease kube-system/cluster-autoscaler
I0903 00:10:15.125576       1 leaderelection.go:278] successfully renewed lease kube-system/cluster-autoscaler
I0903 00:10:16.192639       1 reflector.go:535] k8s.io/client-go/informers/factory.go:134: Watch close - *v1.PersistentVolumeClaim total 0 items received
I0903 00:10:16.795816       1 static_autoscaler.go:231] Starting main loop
I0903 00:10:16.796521       1 aws_manager.go:315] Found multiple availability zones for ASG "eks-test-cas-ng-gpu-d8bdd47d-8232-234e-0544-7d7b3f16519d"; using us-west-2a for failure-domain.beta.kubernetes.io/zone label
I0903 00:10:16.796669       1 aws_manager.go:385] Nodegroup test-cas-ng-gpu is an EKS managed nodegroup.
```

### CODE REVIEW

**Review by gjtempleton — COMMENTED** [review]
Thanks, this mostly looks good to go, one question, and one suggested renaming of a variable for consistency for you.

File: cluster-autoscaler/cloudprovider/aws/aws_manager.go [hunk 1]
```diff
@@ -381,6 +381,11 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
 	// NodeLabels
 	node.Labels = cloudprovider.JoinStringMaps(node.Labels, extractLabelsFromAsg(template.Tags))
 
+	if nodegroupName, ok := node.Labels["nodegroup-name"]; ok {
```
[the remark this answers is not in the document]
Comment (MyannaHarris) [reply cluster-autoscaler/cloudprovider/aws/aws_manager.go:384]: The EKS documentation notes that the "eks:nodegroup-name" tag is an AWS tag that we set, but I can add some documentation in the AWS README here about this as well.

**Comment by gjtempleton** [comment]
Thanks for addressing the feedback!
/lgtm
/approve

**Comment by MyannaHarris** [comment]
Related to https://github.com/kubernetes/autoscaler/pull/3968

**Pull request merged**